patches/bytenode+1.3.7.dev.patch

```diff
@@ -0,0 +1,381 @@
++
++/**
++ * This function runs the compileCode() function (above)
++ * via a child process usine Electron as Node
++ * @param {string} javascriptCode
++ * @returns {Promise<Buffer>} - returns a Promise which resolves in the generated bytecode.
++ */
++const compileElectronCode = function (javascriptCode) {
++  return new Promise((resolve, reject) => {
++    let data = Buffer.from([]);
++
++    const electronPath = path.join(path.dirname(require.resolve('electron')), 'cli.js');
++    if (!fs.existsSync(electronPath)) {
++      throw new Error('Electron not installed');
++    }
++    const bytenodePath = path.join(__dirname, 'cli.js');
++
++    // create a subprocess in which we run Electron as our Node and V8 engine
++    // running Bytenode to compile our code through stdin/stdout
++    const proc = fork(electronPath, [bytenodePath, '--compile', '--no-module', '-'], {
++      env: { ELECTRON_RUN_AS_NODE: '1' },
++      stdio: ['pipe', 'pipe', 'pipe', 'ipc']
++    });
++
++    if (proc.stdin) {
++      proc.stdin.write(javascriptCode);
++      proc.stdin.end();
++    }
++
++    if (proc.stdout) {
++      proc.stdout.on('data', (chunk) => {
++        data = Buffer.concat([data, chunk]);
++      });
++      proc.stdout.on('error', (err) => {
++        console.error(err);
++      });
++      proc.stdout.on('end', () => {
++        resolve(data);
++      });
++    }
++
++    if (proc.stderr) {
++      proc.stderr.on('data', (chunk) => {
++        console.error('Error: ', chunk.toString());
++      });
++      proc.stderr.on('error', (err) => {
++        console.error('Error: ', err);
++      });
++    }
++
++    proc.addListener('message', (message) => console.log(message));
++    proc.addListener('error', err => console.error(err));
++
++    proc.on('error', (err) => reject(err));
++    proc.on('exit', () => { resolve(data); });
++  });
++};
++
++/**
++ * Compiles JavaScript file to .jsc file.
++ * @param   {object|string} args
++ * @param   {string}          args.filename The JavaScript source file that will be compiled
++ * @param   {boolean}         [args.compileAsModule=true] If true, the output will be a commonjs module
++ * @param   {string}          [args.output=filename.jsc] The output filename. Defaults to the same path and name of the original file, but with `.jsc` extension.
++ * @param   {boolean}         [args.electron=false] If true, compile code for Electron (which needs to be installed)
++ * @param   {boolean}         [args.createLoader=false] If true, create a loader file.
++ * @param   {boolean}         [args.loaderFilename='%.loader.js'] Filename or pattern for generated loader files. Defaults to originalFilename.loader.js. Use % as a substitute for originalFilename.
++ * @param   {string}        [output] The output filename. (Deprecated: use args.output instead)
++ * @returns {Promise<string>}        A Promise which returns the compiled filename
++ */
++ const compileFile = async function (args, output) {
```

Comment:
Looks like compileFile & compileElectronCode are 1:1 with the "removed" code in this patch. Can we re-generate to only remove the extra methods? Also why does it matter if we remove the other methods? Any other impact beyond making it a scoped import vs a global?

Reply:
I’ll regenerate this to make sure it’s cleaner. The main reason I want to remove as much as possible is to prevent “dead code” from being in the bundle that could be deleted and new code added in its place.

Reply:
It's used for bundling but then reading the content that was bundled later as well.

Reply:
The require actually includes this code as well, but as I'm thinking through this, since I'm patching anyway I can make this a little clearer: https://github.com/bytenode/bytenode/blob/master/lib/index.js#L252